src/Provider/AbstractProvider.php

```diff
@@ -702,18 +702,26 @@ protected function parseResponse(ResponseInterface $response)
      * Custom mapping of expiration, etc should be done here. Always call the
      * parent method when overloading this method.
      *
-     * @param  mixed $result
-     * @return array
+     * @param  array|string $response
+     * @return array|string
+     * @throws UnexpectedValueException
      */
-    protected function prepareAccessTokenResponse(array $result)
+    protected function prepareAccessTokenResponse($response)
```

Comment:
This change doesn't make sense to me. The OAuth2 spec clearly states what the responses will be, and it is always a hash of keys and values.